content/headless/chunk/option-source-transport.md

```diff
@@ -2,17 +2,18 @@
 ---
 <!-- DISCLAIMER: This file is based on the syslog-ng Open Source Edition documentation https://github.com/balabit/syslog-ng-ose-guides/commit/2f4a52ee61d1ea9ad27cb4f3168b95408fddfdf2 and is used under the terms of The syslog-ng Open Source Edition Documentation License. The file has been modified by Axoflow. -->
 
-## transport()
+## transport() {#transport}
 
-|          |                                                                                                                                                                                     |
-| -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| Type:    | udp, tcp, tls, proxied-tcp, proxied-tls, proxied-tls-passthrough, text-with-nuls |
-| Default: | tcp                                                                                                                                                                                 |
+|          |        |
+| -------- | ------ |
+| Type:    | `auto`, `proxied-tcp`, `proxied-tls`, `proxied-tls-passthrough`, `tcp`, `text-with-nuls`, `tls`, `udp` |
+| Default: | tcp |
 
 *Description:* Specifies the protocol used to receive messages from the source.
 
 For detailed information about how {{% param "product.abbrev" %}} supports the `proxied-tcp`, the `proxied-tls`, and the `proxied-tls-passthrough` parameters, see {{% xref "/chapter-sources/configuring-sources-network/proxy-prot-intro/_index.md" %}}.
 
+`auto`: Available in version 4.10 and later. Automatically detects RFC6587-style octet-count based framing in TCP traffic.
```

Comment:
We may mention or rephrase it in a way that suggests this will be extended in the upcoming releases: auto will basically detect all syslog-protocol-related things: haproxy and even TLS (with an additional security option).